volatility3/framework/plugins/windows/processghosting.py

```diff
@@ -33,52 +35,163 @@ def get_requirements(cls):
-            vollog.warning(
-                "This plugin only supports Windows 10 builds when the ImageFilePointer member of _EPROCESS is present"
+        delete_pending = None
+
+        # if it is 0 then its a side effect of process ghosting
+        if proc.ImageFilePointer.vol.offset != 0:
+            try:
+                file_object = proc.ImageFilePointer
+                delete_pending = file_object.DeletePending
+                file_object = file_object.dereference().vol.offset
+            except exceptions.InvalidAddressException:
+                file_object = 0
+
+        # ImageFilePointer equal to 0 means process ghosting or similar techniques were used
+        else:
+            file_object = 0
+
+        # delete_pending besides 0 or 1 = smear
+        if isinstance(delete_pending, int) and delete_pending not in [0, 1]:
+            vollog.debug(
+                f"Invalid delete_pending value {delete_pending} found for process {proc.UniqueProcessId}"
             )
+            delete_pending = None
+
+        if file_object == 0 or delete_pending == 1:
+            yield file_object, delete_pending, None, proc.SectionBaseAddress
+
+    @classmethod
+    def _vad_checks(
+        cls, control_area: interfaces.objects.ObjectInterface, vad_path: str
+    ) -> Generator[Tuple[int, Optional[int], Optional[int]], None, None]:
+        """
+        Checks the control area for delete on close or delete pending being set
+        """
+        try:
+            file_object = control_area.FilePointer.dereference().cast("_FILE_OBJECT")
+        except exceptions.InvalidAddressException:
             return
 
-        for proc in procs:
-            delete_pending = renderers.UnreadableValue()
-            process_name = utility.array_to_string(proc.ImageFileName)
+        try:
+            delete_on_close = control_area.u.Flags.DeleteOnClose
+        except exceptions.InvalidAddressException:
+            delete_on_close = None
 
-            # if it is 0 then its a side effect of process ghosting
-            if proc.ImageFilePointer.vol.offset != 0:
-                try:
-                    file_object = proc.ImageFilePointer
-                    delete_pending = file_object.DeletePending
-                except exceptions.InvalidAddressException:
-                    file_object = 0
+        if delete_on_close and vad_path.lower().endswith((".exe", ".dll")):
+            yield file_object.vol.offset, None, delete_on_close
 
-            # ImageFilePointer equal to 0 means process ghosting or similar techniques were used
-            else:
-                file_object = 0
+        try:
+            delete_pending = file_object.DeletePending
+        except exceptions.InvalidAddressException:
+            delete_pending = None
 
-            if isinstance(delete_pending, int) and delete_pending not in [0, 1]:
+        if delete_pending and delete_pending == 1:
```

Comment:
This seems like a different check and wouldn't it be accomplished be just `delete_pending == 1` that's kind of a more specific version of `delete_pending`?

Reply:
Yes will just do the == check.

Reply:
addressed in latest commit.